[Jupyter] Prevent error popups when no connected to cluster

bbonenfant opened the pull request:
The polling from the filebrowsers trying to refresh were causing error popups. Switching `auto=false` fixes that issue so that it doesn't automatically load the initial path and start refreshing. The extension then starts the polling automatically when loading/mounting files.

Also added a couple more checks for an `UNKNOWN` state within the poller/frontend.

![image](https://github.com/pachyderm/pachyderm/assets/28938409/4d75cb87-1a06-432e-a89f-1eeba2dd5fe6)

emmajsadams commented on jupyter-extension/src/plugins/mount/customFileBrowser.ts:
```diff
@@ -43,7 +43,7 @@ const createCustomFileBrowser = (
     // if it is then it interferes with the user scrolling immediately.
     state: null,
-    auto: true,
+    auto: false,
```
Maybe add a comment here explaining this issue and why it's important auto: false is set? I am sort of on the fence on if you want to setup a cypress test to make sure the popups don't appear with no config file since as you mentioned it could be pretty tricky. As long as we have a comment here explaining it's important and test the other two parts that seems fine.

emmajsadams commented on jupyter-extension/src/plugins/mount/pollMounts.ts:
```diff
@@ -168,8 +168,8 @@ export class PollMounts {
-        config.cluster_status !== 'INVALID' &&
-        config.cluster_status !== 'VALID_LOGGED_OUT'
+        config.cluster_status === 'VALID_NO_AUTH' ||
+        config.cluster_status === 'VALID_LOGGED_IN'
```
Maybe add a test for this case?

emmajsadams reviewed (approved):
Looks good to me, just some small comments.

bbonenfant commented:
Ok, thank you for the help with figuring out how to get the test working correctly. I made an update to the existing test and cleaned up some of the old cruft in that file which is misleading/no-longer-relevant. 

I added some more assertions within the test I wrote to try to capture both of those testable changes I made. That seemed like an OK place to do this, but let me know if you think that's insufficient.

This should be ready for you to review again.

emmajsadams reviewed (approved):
Looks good to me 👍🏻
